Project Work/controllers/farmer/addCrop.js

```diff
@@ -0,0 +1,68 @@
+const jwt = require('jsonwebtoken');
+const { farmerUser } = require('../../database/models/farmer');
+
+module.exports = async (req, res) => {
+	const token = req.cookies.token;
+	// Check the Existance of Token
+	if (!token) {
+		return res.send({
+			success: false,
+			data: null,
+			error: {
+				code: 1001,
+				msg: "User Does'nt exists",
+			},
+		});
+	}
+
+	// Check if Token is not corrupted
+	try {
+		const data = jwt.verify(token, process.env.SECRET);
+		const userId = data.userId;
+
+		const crop = {
+			storageTime: Date.now(),
+			quantity: 10,
+			warehouseId: 10,
+			paymentId: '12312',
+			cropType: 'Wheat',
+		};
+
+		await farmerUser.findOneAndUpdate(
+			{ _id: userId },
+			{
+				$push: {
```

Comment:
check if warehousrId exits in warehouse database or not

Reply:
Sure! will do and Update it.

Reply:
Done changes.

Reply:
Looks fine to me you can merge it.